patches/api/0418-Bell-API.patch

```diff
@@ -0,0 +1,65 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lama06 <andreasprues36@gmail.com>
+Date: Sun, 22 Jan 2023 11:22:49 +0100
+Subject: [PATCH] Bell API
+
+
+diff --git a/src/main/java/org/bukkit/block/Bell.java b/src/main/java/org/bukkit/block/Bell.java
+index 6fee356a3fc2811d99d9ed2a551118e3e6f0de79..d077bab39dd702b5280455b33db417be4eef6429 100644
+--- a/src/main/java/org/bukkit/block/Bell.java
++++ b/src/main/java/org/bukkit/block/Bell.java
+@@ -3,4 +3,53 @@ package org.bukkit.block;
+ /**
+  * Represents a captured state of Bell.
+  */
+-public interface Bell extends TileState { }
++public interface Bell extends TileState {
++    // Paper start - Bell API
++    /**
++     * Rings the bell in the direction that it is facing.
++     */
++    void ring();
++
++    /**
++     * Rings the bell in the specified direction.
++     * @param direction The direction the bell should ring in
++     * @throws IllegalArgumentException If the specified direction is not {@link BlockFace#isCartesian() cartesian}
++     * or equals either {@link BlockFace#UP} or {@link BlockFace#DOWN}
++     */
++    void ring(@org.jetbrains.annotations.NotNull BlockFace direction);
++
++    /**
++     * Returns if the bell is currently ringing.
++     * @return Whether the bell is currently ringing
++     */
++    boolean isRinging();
++
++    /**
++     * Returns the number of ticks this bell has been ringing for or 0 if it is not ringing.
++     * A bell stops ringing after 50 ticks.
++     * @return The number of ticks this bell has been ringing for
++     */
++    int getTicksRinging();
++
++    /**
++     * Returns the direction the bell is ringing in or null if it is not ringing.
++     * @return The direction the bell is ringing in
++     */
++    @org.jetbrains.annotations.Nullable BlockFace getRingingDirection();
```

Comment:
This method needs a bit more javadocs.
The ringing direction is not really the direction a player rang the bell in, it is the relative block face the bell was hit on.

If I stand infront of a bell that is `facing=west` I can ring it in, as a player, in the directions EAST and WEST.
When I look in the EAST direction and I "ring" the bell, this method returns "WEST" as I hit it while looking east.

If this is a linguistic thing or not (a west wind for example is a wind that blows east etc, so a bell rining from the west to the east could be interpreted as such) does not matter I think, more documentation is required here to ensure we don't run into confusion about the returned value.

Reply:
I have now added javadocs to this method and Bell#ring(BlockFace) to make clear what is meant by the direction a bell is ringing in